src/types.ts

```diff
@@ -36,6 +36,10 @@ export interface HasPreAuthorize<TContextIn, TContextOut> {
   preAuthorize: (context: TContextIn) => TContextOut;
 }
 
```

Comment:
hmm...  Maybe it's better to name this differently than the other OptionallyHas*'s ... like MightHavePreAuthorize
- [ ] misleading name

Reply:
done